### PR TITLE
DSG-1903: Intake form and readonly display routing

### PR DESCRIPTION
**Pull request by kevincheng0**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Added parent component for `transaction/:transactionId` for `IntakeFormComponent` and `ReadonlyComponent` routes. Updated routes as `transaction/:transactionId` and `transaction/:transactionId/readonly` respectively. Accessing a transaction with no `activeTasks` redirects to the readonly. Also updated the existing confirmation page route to `transaction/submitted/:externalTransactionId` as discussed with @davidyoung77 

## Screenshot of the affected changes
image
image

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/public/feature/dashboard/src/lib/lib.routes.ts [hunk 1]
```diff
@@ -15,7 +25,7 @@ export const publicFeatureDashboardRoutes: Route[] = [
   },
   {
     loadComponent: () => import('./components/confirmation').then(module => module.ConfirmationComponent),
-    path: 'submitted/:externalTransactionId',
+    path: 'transaction/submitted/:externalTransactionId',
```
Comment (kevincheng0) [inline libs/public/feature/dashboard/src/lib/lib.routes.ts:28]: @asultan001 David and I discussed about updating the route for the submitted confirmation page, what do you think about this?

**Review by davidyoung77 — COMMENTED** [review]
I have some comments, nice work

**Review by davidyoung77 — APPROVED** [review]
This is good pending confirmation of the submitted route update

File: libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts [hunk 2]
```diff
@@ -13,11 +20,30 @@ import { NuverialBreadcrumbComponent, NuverialButtonComponent, NuverialIconCompo
   templateUrl: './confirmation.component.html',
 })
 export class ConfirmationComponent {
-  public loadApplicationId$ = this._route.snapshot.paramMap.get('externalTransactionId');
+  public loadApplicationId$ = this._route.paramMap.pipe(
+    switchMap(params => {
+      const transactionId = params.get('transactionId') ?? '';
+
+      return this._formTransactionService.loadTransactionDetails$(transactionId);
+    }),
```
Comment (davidyoung77) [inline libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts:28]: If you make this a child component you wont have to loadtransaction details here as that is handled on the parent

File: libs/public/feature/dashboard/src/lib/components/transaction-routing/transaction-routing.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      return this._formTransactionService.loadTransactionDetails$(transactionId);
+    }),
+    catchError(_error => {
+      this._nuverialSnackBarService.notifyApplicationError();
+
+      return EMPTY;
+    }),
+    tap(([_, transactionModel]) => {
+      if (transactionModel.activeTasks.length > 0) {
```
Comment (davidyoung77) [inline libs/public/feature/dashboard/src/lib/components/transaction-routing/transaction-routing.component.ts:34]:
```suggestion
      if (transactionModel.activeTasks.length) {
```

File: libs/public/feature/dashboard/src/lib/lib.routes.ts [hunk 4]
```diff
@@ -15,7 +25,7 @@ export const publicFeatureDashboardRoutes: Route[] = [
   },
   {
     loadComponent: () => import('./components/confirmation').then(module => module.ConfirmationComponent),
-    path: 'submitted/:externalTransactionId',
+    path: 'transaction/:transactionId/submitted',
```
Comment (davidyoung77) [inline libs/public/feature/dashboard/src/lib/lib.routes.ts:28]: This looks like it can be moved to a child route of `transaction/:transactionId`

**Review by davidyoung77 — COMMENTED** [review]
Hey I have a few comments

File: libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts [hunk 5]
```diff
@@ -13,11 +14,15 @@ import { NuverialBreadcrumbComponent, NuverialButtonComponent, NuverialIconCompo
   templateUrl: './confirmation.component.html',
 })
 export class ConfirmationComponent {
-  public loadApplicationId$ = this._route.snapshot.paramMap.get('externalTransactionId');
+  public loadApplicationId$ = this._formTransactionService.transaction$.pipe(
```
Comment (davidyoung77) [inline libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts:17]: Application is on old term that we want to make consistent, can you rename
```suggestion
  public transactionExternalId$ = this._formTransactionService.transaction$.pipe(
```

**Review by davidyoung77 — APPROVED** [review]
thanks I have 1 minor naming comment then this is good

File: libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts (hunk 5 above)
Comment (kevincheng0) [reply libs/public/feature/dashboard/src/lib/components/confirmation/confirmation.component.ts:17]: I renamed it to `externalTransactionId` to keep it consistent with other parts of the app

**Review by davidyoung77 — APPROVED** [review]

**Pull request merged**